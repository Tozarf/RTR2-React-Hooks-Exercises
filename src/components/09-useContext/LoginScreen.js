import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {setUser,user} = useContext(UserContext)

    console.log(user)
    return (
        <>
            <h1>Login Screen</h1>
            <hr/>
            <button
                className="btn btn-primary"
                onClick={()=>setUser({
                    id:1909,
                    name: "Fausto"
                })}
            
            >
                Login
            </button>
        </>
    )
}
