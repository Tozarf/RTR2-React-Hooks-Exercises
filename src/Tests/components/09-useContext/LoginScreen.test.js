import { mount } from 'enzyme'
import React from 'react'
import { LoginScreen } from '../../../components/09-useContext/LoginScreen'
import { UserContext } from '../../../components/09-useContext/UserContext'




describe('Testing <LoginScreen/>', () => {
    const setUser = jest.fn()
    const wrapper = mount(
        <UserContext.Provider value = {{
            setUser
        }}>
            <LoginScreen/>
        </UserContext.Provider>)

    
    test('should match snapshot ', () => {
        

        expect( wrapper ).toMatchSnapshot()

    })

    test('should execute setUser', () => {
        
        wrapper.find("button").prop("onClick")();

        expect (setUser).toHaveBeenCalledWith({
            id:1909,
            name: "Fausto"
        })
       
    })
    
    
})
