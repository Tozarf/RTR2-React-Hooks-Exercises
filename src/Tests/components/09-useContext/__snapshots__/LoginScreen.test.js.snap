// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <LoginScreen/> should match snapshot  1`] = `
Array [
  <h1>
    Login Screen
  </h1>,
  <hr />,
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
